### PR TITLE
Fix: Odometry fuser Z axis

### PR DESCRIPTION
**Pull request by Flova**
This pull request mostly removes the usage of Euler coordinates from the odometry fuser.
This also implements the rot_conv library for this task.

## Proposed changes
Use quaternions for yaw removal with the rot_conv library.

## Related issues
#129 

## Necessary checks
- [ ] Update package version
- [x] Run linters
- [x] Run `catkin build`
- [x] Write documentation
- [x] Test on your machine
- [x] Test on the robot

### CODE REVIEW

File: bitbots_odometry/src/odometry_fuser.cpp [hunk 1]
```diff
@@ -63,13 +65,11 @@ OdometryFuser::OdometryFuser() : tf_listener_(tf_buffer_) {
     }
 
     if (imu_active || odom_active) {
-      double placeholder, imu_roll, imu_pitch, walking_yaw;
-
+      double walking_yaw, placeholder;
       // get roll an pitch from imu
       tf2::Quaternion imu_orientation;
       tf2::fromMsg(_imu_data.orientation, imu_orientation);
       tf2::Matrix3x3 imu_rotation_matrix(imu_orientation);
```
Comment (SammyRamone) [inline bitbots_odometry/src/odometry_fuser.cpp]: is imu_rotation_matrix still used?

File: bitbots_odometry/src/odometry_fuser.cpp [hunk 2]
```diff
@@ -1,4 +1,6 @@
 #include <bitbots_odometry/odometry_fuser.h>
+#include <rot_conv/rot_conv.h>
+#include <Eigen/Geometry>
```
Comment (SammyRamone) [inline bitbots_odometry/src/odometry_fuser.cpp]: includes should be in header

Comment (Flova) [reply bitbots_odometry/src/odometry_fuser.cpp]: Okay

File: bitbots_odometry/src/odometry_fuser.cpp (hunk 1 above)
Comment (Flova) [reply bitbots_odometry/src/odometry_fuser.cpp]: No, it isnt used anymore. Thanks.

**Pull request merged**